### DISCUSSION TITLE
Helix crates

### DISCUSSION THREAD

**Discussion by matoous** [helix-editor/helix#10074] [Q&A] [answered]
Hi!

Going through the codebase after a long time and I have a few questions: what's the current difference/purpose of `helix-tui`, `helix-term`, and `helix-view`? And maybe also follow-up question - what's the _desired_ state?

To my understanding (append question mark after everything):

- `helix-term` is the main entrypoint for the editor and it's where commands and features live. One could say the "business logic" of helix.
- `helix-tui` provides terminal UI components
- `helix-view` provides abstractions over UI components, so in theory and far future, one should be able to plug either TUI or GUI behind `helix-view`

Looking at the current state, the difference between `helix-view` and `helix-tui` isn't fully clear to me. Furthermore, if I am not missing anything, `helix-term` itself has a lot of TUI rendering logic.

Could somebody please shed a light on this topic? 🙏

**Comment by pascalkuthe** [comment, marked as answer]
`helix-tui` is a fork of `tui-rs` so it's a general term rendering libtary/abstraction.


`helix-view` and `helix-term` are a complex subject and we want to move most things to `helix-view` (#5581). The only thing that will remain in `helix-term` is the rendering implementation for various components for tui (so there could be a seperate helix-gui crate that does the same thing for a gui with as little duplication as possible). 

°helid-view` ans `helix-term` are also kind of a a kitchen skin right now. I want to split them up into more self-contained crates. That is all in the more distant future, helix needs lots of architectural work
